src/Maybe/Extra.elm

```diff
@@ -145,6 +145,42 @@ filter f m =
             Nothing
 
 
+{-| Put a value in a maybe dependig on a boolean
+
+    toMaybe True 1
+    --> Nothing
+
+    toMaybe False 1
+    --> Nothing
+
+-}
+toMaybe : Bool -> a -> Maybe a
+toMaybe test value =
+    if test then
+        Just value
+
+    else
+        Nothing
+
+
+{-| Put a value in a Maybe depending on a predicate
+
+    guarded ((<) 10) 5
+    --> Nothing
+
+    guarded ((<) 2) 5
```

Comment:
I always have trouble reading infix functions. In my head I read `((<) 2)` as "is less than 2" instead of "2 is less than it". Is there a boolean function to use as an example that doesn't carry the same risk of switching it around?

One function that I'm likely to use in in the real world might be
```
guarded (not << String.isEmpty) ""
--> Nothing

guarded (not << String.isEmpty) "not empty"
--> Just "not empty"
```

Another possibility that might be easier to read could be
```
guarded (String.contains "a") "cat"
--> Just "cat"

guarded (String.contains "a") "dog"
--> Nothing
```

Thoughts?

Reply:
I agree that the infix operator is hard to understand; which in turn makes it hard to know if `True` or `False` leads to `Just 5` or `Nothing`.

Reply:
Agreed. I've used @skyqrose first suggestion. I considered using both but maybe one is enough to illustrate the point?